includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -88,7 +89,44 @@ function get_columns() {
 		$columns = apply_filters( 'sensei_analysis_overview_columns', $columns, $this );
 		return $columns;
 	}
-
+	/**
+	 * Append the count value to column headers where applicable
+	 *
+	 * @since  4.2.0
+	 * @access private
+	 *
+	 * @param array $columns Array of columns for the report table.
+	 * @return array The array of columns to use with the table with columns appended to their title
+	 */
+	public function add_totals_to_report_column_headers( array $columns ) {
+		remove_filter( 'sensei_analysis_overview_columns', array( $this, __FUNCTION__ ) );
```

Comment:
Just to note that a filter/action removing itself is an anti-pattern and you should look for other ways around if possible. Calling your filter multiple times should be solved by https://github.com/Automattic/sensei/commit/bfc499294d6f5614a33e7e5ffa7633bc43be6f22 so you could duplicate that solution, rebase on top of that branch or wait for it to be merged.